shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -2125,4 +2127,12 @@ object GlutenConfig {
         "Otherwise, throw an exception.")
       .booleanConf
       .createWithDefault(true)
+
+  val CAST_FROM_VARCHAR_ADD_TRIM_NODE =
+    buildConf("spark.gluten.velox.castFromVarcharAddTrimNode")
+      .internal()
+      .doc("If true, will add a trim node which has the same sementic as vanilla Spark." +
+        "Otherwise, do nothing.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
I think we can set default value to false, because all GHA has passed and performance boost can not be ignored. If incorrect result issue occurs, we can fix it or make this config more fine-grained. 

Just like scan parquet complex types, we remove force fallback config after all UT passed.

@PHILO-HE @jackylee-ch

Reply:
It's ok to me. And if it's set default to false, we need add this config to `Configuration.md` to make sure users know about it.

Reply:
> need add this config to

I'll do this.

Reply:
> It's ok to me. And if it's set default to false, we need add this config to `Configuration.md` to make sure users know about it.

I'll do this.